nemo/core/classes/modelPT.py

```diff
@@ -204,6 +205,8 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
 
         # Setup nsys profiling if it has been enabled in the model config
         self._setup_profiling()
+        # real accurate _batch_idx. We found that the `batch_idx` in `on_train_batch_start` and `on_train_batch_end` has a bug.
+        self._real_batch_idx = 0
```

Comment:
Can this be avoided by instead using self.trainer.global_step instead of trainer is not None (and it's never none during training step call anyway)